uaclient/exceptions.py

```diff
@@ -419,59 +412,28 @@ def __init__(
         )
 
 
-class ContractAPIError(UrlError):
-    def __init__(self, e, error_response):
-        super().__init__(e, e.code, e.headers, e.url)
-
-        if error_response is None:
-            error_response = {}
-
-        self.api_error = error_response
-
-    def __contains__(self, error_code):
-        if error_code == self.api_error.get("code"):
-            return True
-        if self.api_error.get("message", "").startswith(error_code):
-            return True
-
-        return False
-
-    def __get__(self, error_code, default=None):
-        if self.api_error.get("code") == error_code:
-            return self.api_error.get("message")
-        return default
-
-    def __str__(self):
-        prefix = super().__str__()
-        return (
-            prefix
-            + ": ["
-            + str(self.url)
-            + "]"
-            + ", "
-            + self.api_error.get("message", "")
+class ExternalAPIError(UserFacingError):
+    def __init__(self, url: str, code: int, body: str):
+        self.url = url
+        self.code = code
+        self.body = body
+        super().__init__(
+            "Error connecting to {}: {} {}".format(
```

Comment:
Should this message sit on the `messages` module ?

Reply:
Yeah, I'll move it